### PR TITLE
CPAT: Added option to use reference genome from builtin references

### PR DESCRIPTION
**Pull request by SaimMomin12**
Based on a request from a Galaxy user regarding feature addition in the CPAT tool as described in #1458 this PR adds the functionality of utilizing built in reference genomes from Galaxy as well as retains the functionality of using reference genome from history.

### CODE REVIEW

**Comment by SaimMomin12** [comment]
That skipped out of my mind, sure I will add it. Thanks :)

File: tools/cpat/cpat.xml [hunk 1]
```diff
@@ -144,6 +166,41 @@
             </output>
             <output name="no_orf_seqs" file="test02_no_orgs.txt" ftype="txt"/>
         </test>
+        <test expect_num_outputs="4">
+            <param name="gene" value="sequences.fasta.gz"/>
+            <conditional name="ref_source">
+                <param name="source" value="builtin" />
+                <param name="ref_fasta" value="test1.fasta.gz" />
```
Comment (bgruening) [inline tools/cpat/cpat.xml:173]: ref_fasta_builtin I assume

Comment (SaimMomin12) [reply tools/cpat/cpat.xml:173]: I have defined the value as `builtin` as below

```
<param type="select" label="Reference genome source" name="source">
    <option value="history" selected="true">Use from History</option>
    <option value="builtin">Use Built-in</option>
</param>
```

Comment (bgruening) [reply tools/cpat/cpat.xml:173]: But ref_fasta is not a Param in the built-in case isn't it? On my mobile, no maybe I don't see it

Comment (SaimMomin12) [reply tools/cpat/cpat.xml:173]: Yes I checked it, it should be ref_fasta_builtin I have made the change in recent commit

File: tools/cpat/test-data/all_fasta.loc
Comment (bgruening) [inline tools/cpat/test-data/all_fasta.loc:1]: its better no name this one `all_fasta.loc.test`

File: tools/cpat/tool_data_table_conf.xml.test [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<tables>
+    <!-- Locations of reference genome files in fasta format -->
+    <table name="all_fasta" comment_char="#">
+        <columns>value, name, path</columns>
+        <file path="${__HERE__}/test-data/all_fasta.loc" />
```
Comment (bgruening) [inline tools/cpat/tool_data_table_conf.xml.test:6]:
```suggestion
        <file path="${__HERE__}/test-data/all_fasta.loc.test" />
```

**Review by bgruening — APPROVED** [review]
This looks all great now. The only thing that worries me is - why did the test pass in the first place (before your last commit), they shouldn't or? I'm just worried the test is not stringent enough, but will trust you if you think everything is good.

**Comment by SaimMomin12** [comment]
> This looks all great now. The only thing that worries me is - why did the test pass in the first place (before your last commit), they shouldn't or? I'm just worried the test is not stringent enough, but will trust you if you think everything is good.

I think that was merely due to the less stringency of the test params. I think everything is good to go. This can be merged.

**Pull request merged**